src/resolver.cpp

```diff
@@ -21,103 +27,176 @@ TF_DEBUG_CODES(OPENASSETIO_RESOLVER)
 
 PXR_NAMESPACE_CLOSE_SCOPE
 
+namespace {
+/// Converter logger from OpenAssetIO log framing to USD log outputs.
+class UsdOpenAssetIOResolverLogger : public openassetio::log::LoggerInterface {
+ public:
+  void log(Severity severity, const openassetio::Str &message) override {
+    switch (severity) {
+      case Severity::kCritical:
+        TF_ERROR(TfDiagnosticType::TF_DIAGNOSTIC_FATAL_ERROR_TYPE, message + "\n");
+        break;
+      case Severity::kDebug:
+      case Severity::kDebugApi:
+        TF_DEBUG(OPENASSETIO_RESOLVER).Msg(message + "\n");
+        break;
+      case Severity::kError:
+        // TODO(EM) : Review to see which error types are most appropriate,
+        //  are all errors (not criticals) non fatal?
+        TF_ERROR(TfDiagnosticType::TF_DIAGNOSTIC_NONFATAL_ERROR_TYPE, message + "\n");
+        break;
+      case Severity::kInfo:
+      case Severity::kProgress:
+        TF_INFO(OPENASSETIO_RESOLVER).Msg(message + "\n");
+        break;
+      case Severity::kWarning:
+        TF_WARN(TfDiagnosticType::TF_DIAGNOSTIC_WARNING_TYPE, message + "\n");
+        break;
+    }
+  }
+};
+
+class UsdOpenAssetIOHostInterface : public openassetio::hostApi::HostInterface {
+ public:
+  [[nodiscard]] openassetio::Identifier identifier() const override {
+    return "org.openassetio.usdresolver";
+  }
+
+  [[nodiscard]] openassetio::Str displayName() const override {
+    return "OpenAssetIO USD Resolver";
+  }
+};
+}  // namespace
+
 // ------------------------------------------------------------
 /* Ar Resolver Implementation */
 UsdOpenAssetIOResolver::UsdOpenAssetIOResolver() {
-  TF_DEBUG(OPENASSETIO_RESOLVER).Msg("OPENASSETIO_RESOLVER: " + TF_FUNC_NAME() + "\n");
+  logger_ =
+      openassetio::log::SeverityFilter::make(std::make_shared<UsdOpenAssetIOResolverLogger>());
+
+  auto managerImplementationFactory =
+      openassetio::python::hostApi::createPythonPluginSystemManagerImplementationFactory(logger_);
+
+  const auto hostInterface = std::make_shared<UsdOpenAssetIOHostInterface>();
+
+  manager_ = openassetio::hostApi::ManagerFactory::defaultManagerForInterface(
+      hostInterface, managerImplementationFactory, logger_);
+
+  if (!manager_) {
+    throw std::invalid_argument{
+        "OpenAssetIO manger plugin not found. Check OPENASSETIO_DEFAULT_CONFIG is set."};
```

Comment:
[fix] Can we re-word this slightly (and use the static) as it could be interpreted that a configured plugin was missing, which would actually surface as an execption in `defaultManagerForInterface` iirc. We usually say:

https://github.com/OpenAssetIO/OpenAssetIO/blob/fbc24bbeb32f6aa0c43cc985310364fa3816bc30/examples/host/simpleResolver/simpleResolver.py#L142

Reply:
Took the same pattern, this what you were thinking?

https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/16/commits/87ca72d3a5937f0bf3245513dd0fe536eb05f3b1